rp_transferto/tests/test_tasks.py

```diff
@@ -2,59 +2,107 @@
 from unittest.mock import MagicMock
 from django.test import TestCase, override_settings
 
-from temba_client.v2 import TembaClient
-
 from rp_transferto.tasks import topup_data
-from rp_transferto.utils import TransferToClient, TransferToClient2
+from rp_transferto.models import MsisdnInformation
 
 from .constants import (
     MSISDN_INFO_RESPONSE_DICT,
     GET_PRODUCTS_RESPONSE_DICT,
     POST_TOPUP_DATA_RESPONSE,
 )
 
-fake_get_misisdn_info = MagicMock(return_value=MSISDN_INFO_RESPONSE_DICT)
-fake_get_operator_products = MagicMock(return_value=GET_PRODUCTS_RESPONSE_DICT)
-fake_topup_data = MagicMock(return_value=POST_TOPUP_DATA_RESPONSE)
-fake_get_contacts_query = MagicMock()
-fake_get_contacts_query.first.return_value = {}
-fake_update_contact = MagicMock(return_value={})
-
 
-@patch.object(TransferToClient, "get_misisdn_info", fake_get_misisdn_info)
-@patch.object(
-    TransferToClient2, "get_operator_products", fake_get_operator_products
-)
-@patch.object(TransferToClient2, "topup_data", fake_topup_data)
-@patch.object(TembaClient, "get_contacts", fake_get_contacts_query)
-@patch.object(TembaClient, "update_contact", fake_update_contact)
+@patch("temba_client.v2.TembaClient.update_contact")
+@patch("temba_client.v2.TembaClient.get_contacts")
+@patch("rp_transferto.utils.TransferToClient2.topup_data")
+@patch("rp_transferto.utils.TransferToClient2.get_operator_products")
+@patch("rp_transferto.utils.TransferToClient.get_misisdn_info")
 @override_settings(
     TRANSFERTO_LOGIN="fake_transferto_login",
     TRANSFERTO_TOKEN="fake_transferto_token",
     TRANSFERTO_APIKEY="fake_transferto_apikey",
     TRANSFERTO_APISECRET="fake_transferto_apisecret",
 )
 class TestTopupDataTask(TestCase):
-    def setUp(self):
-        pass
+    def test_successsful_run(
```

Comment:
Doesn't look like this test is checking anything?

Reply:
This test checks that in the case where we don't have information stored in the db, so it calls the Task and then check that all of the functions were called accordingly. See line no 57 where the task is actually run.